code-challenges/fizzBuzzTree/fizz-buzz-tree.js

```diff
@@ -0,0 +1,76 @@
+class Node {
+  constructor (value) {
+    this.value = value
+    this.left = null
+    this.right = null
+  }
+}
+
+class BinaryTree {
+  constructor () {
+    this.root = null
+  }
+
+  preOrder () {
+    const output = []
+    const traverse = (Node) => {
+      if (!Node) {
+        return
+      }
+      output.push(Node.value)
+      if (Node.left) {
+        traverse(Node.left)
+      }
+      if (Node.right) {
+        traverse(Node.right)
+      }
+    }
+    traverse(this.root)
+    return output
+  }
+
+  // from Cait, for constructing BinaryTree
+  insert (value) {
+    const newNode = new Node(value)
+    if (this.root === null) {
+      return this.root = newNode
+    }
+    var current = this.root
+    while (true) {
+      if (value === current.value) return undefined
+      if (value < current.value) {
+        if (current.left === null) {
+          return current.left = newNode
+        }
+        current = current.left
+      } else {
+        if (current.right === null) {
+          return current.right = newNode
+        }
+        current = current.right
+      }
+    }
+  }
+}
+
+function traverse (Node) {
```

Comment:
I like that you put traverse function outside the BinaryTree class.
'traverse' is a name for tree traversal (not related to fizzbuzz). Can you come up with a better name for this function?